core/state/journal.go

```diff
@@ -396,20 +418,21 @@ func (ch codeChange) copy() journalEntry {
 	}
 }
 
-func (ch storageChange) revert(s *StateDB) {
+func (ch *storageChange) revert(s *StateDB) {
 	s.getStateObject(ch.account).setState(ch.key, ch.prevvalue, ch.origvalue)
 }
 
-func (ch storageChange) dirtied() *common.Address {
+func (ch *storageChange) dirtied() *common.Address {
 	return &ch.account
 }
 
-func (ch storageChange) copy() journalEntry {
-	return storageChange{
-		account:   ch.account,
-		key:       ch.key,
-		prevvalue: ch.prevvalue,
-	}
+func (ch *storageChange) copy() journalEntry {
+	cpy := storageChangePool.Get().(*storageChange)
+	cpy.account = ch.account
+	cpy.key = ch.key
+	cpy.prevvalue = ch.prevvalue
+	cpy.origvalue = ch.origvalue
```

Comment:
we may keep the beheavior as before:
```
cpy.origvalue = common.Hash{}
```

Reply:
Agree, It's a bug.
I have created a PR to feed back to go-ethereum, so they will double check this
https://github.com/ethereum/go-ethereum/pull/31874